python/pipeline/meso.py

```diff
@@ -519,55 +565,301 @@ def make(self, key):
+                fps = (ScanInfo & key).fetch1("fps")
+                window_size = np.max((int(fps / 5), 3))
+            else:
+                raise PipelineException(f"The smoothing for the second iteration of motion correction is not defined for {key['motion_correction_method']}")
+
+            # In place low memory convolution. This is slower
+            # but allows for the entire scan to be in memory.
+            start_indices = np.arange(0, scan.shape[-1], 1000)
+            if start_indices[-1] != scan.shape[-1]:
+                start_indices = np.insert(
+                    start_indices, len(start_indices), scan.shape[-1]
+                )
+            for start_idx, end_idx in tqdm(
+                zip(start_indices[:-1], start_indices[1:]), total=len(start_indices) - 1
+            ):
+                conv_start = np.max((0, start_idx - window_size))
+                conv_end = np.min((end_idx + window_size, scan.shape[-1]))
+                conv_fragment = ndimage.convolve1d(
+                    scan[:, :, conv_start:conv_end],
+                    np.ones(window_size) / window_size,
+                    axis=-1,
+                    mode="reflect",
+                )
+
+                # Use these so the offset will be zero when at beginning or end of scan
+                slice_start = start_idx - conv_start
+                slice_end = end_idx - conv_end + conv_fragment.shape[-1]
+                scan[:, :, start_idx:end_idx] = conv_fragment[
+                    :, :, slice_start:slice_end
+                ]
+
+            scan = np.expand_dims(scan, axis=(0, 3))
+            template = galvo_corrections.create_refined_template(scan, x_shifts, y_shifts, key)
+
+            f = performance.parallel_motion_shifts  # function to map
+            kwargs = {
+                "raster_phase": raster_phase,
+                "fill_fraction": fill_fraction,
+                "template": template,
+            }
+            results = performance.map_frames(
+                f,
+                scan,
+                field_id=0,
+                y=slice(skip_rows, -skip_rows),
+                x=slice(skip_cols, -skip_cols),
+                channel=0,
+                kwargs=kwargs,
+            )
+
+            second_y_shifts = np.zeros(scan.shape[-1])
+            second_x_shifts = np.zeros(scan.shape[-1])
+            for frames, chunk_y_shifts, chunk_x_shifts in results:
+                second_y_shifts[frames] = chunk_y_shifts
+                second_x_shifts[frames] = chunk_x_shifts
+
+            second_x_shifts, second_y_shifts, _ = galvo_corrections.process_xy_motion(
+                second_x_shifts,
+                second_y_shifts,
+                key,
+            )
+
+            x_shifts = second_x_shifts + old_x_shifts
+            y_shifts = second_y_shifts + old_y_shifts
+
+        # Run a second iteration of motion correction using multiple local templates
```

Comment:
Is it possible to consolidate this code with the above code, since it seems to be heavily redundant?